alerts/resource_alerts.libsonnet

```diff
@@ -25,65 +25,65 @@
+              description: 'Cluster {{ $labels.%(clusterLabel)s }} has overcommitted CPU resource requests for Pods by {{ $value }} CPU shares and cannot tolerate node failure.' % $._config,
               summary: 'Cluster has overcommitted CPU resource requests.',
             },
             'for': '10m',
           },
           {
             alert: 'KubeMemoryOvercommit',
             expr: |||
-              sum(namespace_memory:kube_pod_container_resource_requests:sum{%(ignoringOverprovisionedWorkloadSelector)s}) - (sum(kube_node_status_allocatable{resource="memory"}) - max(kube_node_status_allocatable{resource="memory"})) > 0
+              sum(namespace_memory:kube_pod_container_resource_requests:sum{%(ignoringOverprovisionedWorkloadSelector)s}) by (%(clusterLabel)s) - (sum(kube_node_status_allocatable{resource="memory"}) by (%(clusterLabel)s) - max(kube_node_status_allocatable{resource="memory"}) by (%(clusterLabel)s)) > 0
               and
-              (sum(kube_node_status_allocatable{resource="memory"}) - max(kube_node_status_allocatable{resource="memory"})) > 0
+              (sum(kube_node_status_allocatable{resource="memory"}) by (%(clusterLabel)s) - max(kube_node_status_allocatable{resource="memory"}) by (%(clusterLabel)s)) > 0
             ||| % $._config,
             labels: {
               severity: 'warning',
             },
             annotations: {
-              description: 'Cluster has overcommitted memory resource requests for Pods by {{ $value | humanize }} bytes and cannot tolerate node failure.',
+              description: 'Cluster {{ $labels.%(clusterLabel)s }} has overcommitted memory resource requests for Pods by {{ $value | humanize }} bytes and cannot tolerate node failure.' % $._config,
               summary: 'Cluster has overcommitted memory resource requests.',
             },
             'for': '10m',
           },
           {
             alert: 'KubeCPUQuotaOvercommit',
             expr: |||
-              sum(min without(resource) (kube_resourcequota{%(prefixedNamespaceSelector)s%(kubeStateMetricsSelector)s, type="hard", resource=~"(cpu|requests.cpu)"}))
+              sum(min without(resource) (kube_resourcequota{%(prefixedNamespaceSelector)s%(kubeStateMetricsSelector)s, type="hard", resource=~"(cpu|requests.cpu)"})) by (%(clusterLabel)s)
                 /
-              sum(kube_node_status_allocatable{resource="cpu", %(kubeStateMetricsSelector)s})
+              sum(kube_node_status_allocatable{resource="cpu", %(kubeStateMetricsSelector)s}) by (%(clusterLabel)s)
                 > %(namespaceOvercommitFactor)s
             ||| % $._config,
             labels: {
               severity: 'warning',
             },
             annotations: {
-              description: 'Cluster has overcommitted CPU resource requests for Namespaces.',
+              description: 'Cluster {{ $labels.%(clusterLabel)s }}  has overcommitted CPU resource requests for Namespaces.' % $._config,
```

Comment:
I'm not sure whether it is a good idea to add `clusterLabel` as part of alert description, all the labels must be already part of the alert data right?

Reply:
Yes, the labels are present on the alert, but that requires that a you go view the alert in prometheus to ascertain which cluster is affected in a multi-cluster environment.

When you receive a notification (email, slack, etc) triggered off of an alert, without the cluster name it is ambiguous.

Also, the [mixtool guidelines](https://monitoring.mixins.dev/#guidelines-for-alert-names-labels-and-annotations) advise that you templatize the description to make it contextually "complete" for a given alert.

Reply:
May be we could avoid adding extra space when there is no clusterLabel configured.

Reply:
Ahh yes, good call. We primarily use the mixin in mult-cluster mode and thus always have a cluster label.

There are some related PRs that would need adjusting as well.

I'll make this change.

Reply:
@arajkumar and @paulfantom This is now updated to properly support both multi-cluster and single cluster scenarios.

In retrospect, the changes made in #766 and #704 need a similar if/else treatment. It may actually make sense to have completely separate files, or config for multi-cluster vs. single cluster. 🤔